File: backend/routers/special_days.py
from fastapi import APIRouter
from routers.important_days import (
    get_important_days as get_special_days,
    get_today_important_days as get_today_special_days,
    get_important_day_by_id as get_special_day_by_id,
    create_important_day as create_special_day,
    update_important_day as update_special_day,
    delete_important_day as delete_special_day,
    generate_wish,
    send_test_email
)

router = APIRouter(tags=["Special Days"])

router.add_api_route("", get_special_days, methods=["GET"])
router.add_api_route("/today", get_today_special_days, methods=["GET"])
router.add_api_route("/generate-wish", generate_wish, methods=["POST"])
router.add_api_route("/send-test-email", send_test_email, methods=["POST"])
router.add_api_route("/{important_day_id}", get_special_day_by_id, methods=["GET"])
router.add_api_route("", create_special_day, methods=["POST"], status_code=201)
router.add_api_route("/{important_day_id}", update_special_day, methods=["PUT"])
router.add_api_route("/{important_day_id}", delete_special_day, methods=["DELETE"], status_code=204)

File: web/src/services/special-days.ts
import api from "@/lib/axios";
import { SpecialDay, ScheduledEmail } from "@/types/SpecialDay";

export const specialDaysService = {
  getSpecialDays: async (params?: { skip?: number; limit?: number }) => {
    const response = await api.get<SpecialDay[]>("/api/important-days", { params });
    return response.data;
  },

  getTodaySpecialDays: async () => {
    const response = await api.get<SpecialDay[]>("/api/important-days/today");
    return response.data;
  },

  getSpecialDay: async (id: number) => {
    const response = await api.get<SpecialDay>(`/api/important-days/${id}`);
    return response.data;
  },

  createSpecialDay: async (data: Partial<SpecialDay> & { schedule_for_tomorrow?: boolean }) => {
    const response = await api.post<SpecialDay>("/api/important-days", data);
    return response.data;
  },

  updateSpecialDay: async (id: number, data: Partial<SpecialDay> & { schedule_for_tomorrow?: boolean }) => {
    const response = await api.put<SpecialDay>(`/api/important-days/${id}`, data);
    return response.data;
  },

  deleteSpecialDay: async (id: number) => {
    await api.delete(`/api/important-days/${id}`);
  },

  generateWish: async (data: { type: string; person_name: string; custom_type?: string | null }) => {
    const response = await api.post<{ subject: string; message: string }>("/api/important-days/generate-wish", data);
    return response.data;
  },

  sendTestEmail: async (data: { recipient_email: string; email_subject?: string; email_message?: string }) => {
    const response = await api.post<{ message: string }>("/api/important-days/send-test-email", data);
    return response.data;
  },

  getScheduledEmails: async () => {
    const response = await api.get<ScheduledEmail[]>("/api/scheduled-emails");
    return response.data;
  },

  deleteScheduledEmail: async (id: number) => {
    await api.delete(`/api/scheduled-emails/${id}`);
  },
};
